### ISSUE TITLE
Air gap single node installation not Provisioning

### ISSUE THREAD

**Issue by turicumoperarius** [rancher/rancher#21241]
After installing rachner by using the air gap single node instruction I was not able to provision a cluster on the same node. I used the same method ("Custom" - "All Node Roles on the same Node") like I used multiple times with succsess on non air gapped single node systems.

Additinally to the non air gapped installatoins I provided the nodes local ip address by using "--address <LocalIP>" on the docker command to start the provisioning of the node. It seems that everything seems to start fine but one container is getting into a restarting loop with the error:

RTNETLINK answers: Network is unreachable
ERROR: -- --address is a required option

By inspecting this containers arguments the "--address" argument is missing. The container with the restarting behaviour is using the "rancher-agent" image and the other additional arguments which are passed to "run.sh" are "--no-register" and "--only-write-certs".

I'm using rancher version v2.2.4 on rancheros v1.5.2.

**Comment by superseb** [comment]
Can you provide `docker inspect` and `docker logs` output for the containers? As well as the output of `ip addr show` from the host so we can reopen and investigate.

**Comment by turicumoperarius** [comment]
Here are the requested logs. I used rancher v2.2.7 and rancheros v1.5.2 for the test.

I used this docker command to initiate the rancher agent:

`sudo docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run 192.168.10.85:5000/rancher/rancher-agent:v2.2.7 --server https://192.168.10.85:8443 --token 29nstxklzc99lr7nv4lkbb76j8km7htc45zcqghspq5swnlpp8g5tt --ca-checksum f78d9e4788bb792029aa09d083ff8e93cf8b87b043e174c2dbdd9bb86d616e21 --address 192.168.10.238 --etcd --controlplane --worker`

For me i think the problem lays at the call of the container "inspiring_lamarr" which seems to miss the "--address" argument.


[rancher info.txt](https://github.com/rancher/rancher/files/3480499/rancher.info.txt)

**Comment by superseb** [comment]
Ok, I see the where this is coming from. I tested the situation and my cluster still provisions fine, the error in your etcd container shows that the machine is not properly cleaned before being added (leftover certificates are being used). Can you clean the machine according to https://rancher.com/docs/rancher/v2.x/en/cluster-admin/cleaning-cluster-nodes/ and try again?

I can still fix this behavior but its not blocking provisioning in my setup (v2.2.7 with one node with all roles and one as worker)

**Comment by turicumoperarius** [comment]
Thanks for the tipp. To make it as easy and quick as possible, i started with a fresh installtion of rancheros v1.5.3 again for the node. Now it seems, that etcd is working, but it seems to stuck at the kubelet distribution.

At Clusterlevel i get the status message:

`Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system`

At the node level the following message is displayed:

`Waiting to register with Kubernetes`

My docker "ps-a" command is now showing a lot more containers than previous.
 
```
CONTAINER ID        IMAGE                                                     COMMAND                  CREATED             STATUS                      PORTS               NAMES
e63c60795ddc        192.168.10.85:5000/rancher/rke-tools:v0.1.40              "sh -c 'mkdir -p /va…"   2 seconds ago       Up Less than a second       80/tcp              rke-log-linker
6320fc712f3e        3ce97e5c7e3a                                              "kubectl apply -f /e…"   17 seconds ago      Exited (1) 16 seconds ago                       k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-ldp9j_kube-system_549758db-b9ea-11e9-b44d-005056affe3f_0
bf5889e6a0f6        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 18 seconds ago      Exited (0) 15 seconds ago                       k8s_POD_rke-network-plugin-deploy-job-ldp9j_kube-system_549758db-b9ea-11e9-b44d-005056affe3f_0
5513cff42ba0        3ce97e5c7e3a                                              "kubectl apply -f /e…"   5 minutes ago       Exited (1) 5 minutes ago                        k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-6d925_kube-system_95d7e882-b9e9-11e9-b44d-005056affe3f_0
1944b913fceb        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 5 minutes ago       Exited (0) 5 minutes ago                        k8s_POD_rke-network-plugin-deploy-job-6d925_kube-system_95d7e882-b9e9-11e9-b44d-005056affe3f_0
615bddfab6f2        3ce97e5c7e3a                                              "kubectl apply -f /e…"   8 minutes ago       Exited (1) 8 minutes ago                        k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-jrdzf_kube-system_367659df-b9e9-11e9-b44d-005056affe3f_0
45e42e96fe1a        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 8 minutes ago       Exited (0) 8 minutes ago                        k8s_POD_rke-network-plugin-deploy-job-jrdzf_kube-system_367659df-b9e9-11e9-b44d-005056affe3f_0
dc08ccd99212        3ce97e5c7e3a                                              "kubectl apply -f /e…"   9 minutes ago       Exited (1) 9 minutes ago                        k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-qqslh_kube-system_06c20818-b9e9-11e9-b44d-005056affe3f_0
4aeb6b7c0b04        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 9 minutes ago       Exited (0) 9 minutes ago                        k8s_POD_rke-network-plugin-deploy-job-qqslh_kube-system_06c20818-b9e9-11e9-b44d-005056affe3f_0
4b0b746f3b1b        3ce97e5c7e3a                                              "kubectl apply -f /e…"   10 minutes ago      Exited (1) 10 minutes ago                       k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-l642r_kube-system_eee787c6-b9e8-11e9-b44d-005056affe3f_0
fa0102b5a6ee        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 10 minutes ago      Exited (0) 10 minutes ago                       k8s_POD_rke-network-plugin-deploy-job-l642r_kube-system_eee787c6-b9e8-11e9-b44d-005056affe3f_0
8bc5b0f07d4b        3ce97e5c7e3a                                              "kubectl apply -f /e…"   10 minutes ago      Exited (1) 10 minutes ago                       k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-jskfw_kube-system_e2f8de57-b9e8-11e9-b44d-005056affe3f_0
72d79bfab285        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 10 minutes ago      Exited (0) 10 minutes ago                       k8s_POD_rke-network-plugin-deploy-job-jskfw_kube-system_e2f8de57-b9e8-11e9-b44d-005056affe3f_0
17aea9964d2a        3ce97e5c7e3a                                              "kubectl apply -f /e…"   10 minutes ago      Exited (1) 10 minutes ago                       k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-dcgbp_kube-system_dd000cf7-b9e8-11e9-b44d-005056affe3f_0
1be8febf4fda        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 10 minutes ago      Exited (0) 10 minutes ago                       k8s_POD_rke-network-plugin-deploy-job-dcgbp_kube-system_dd000cf7-b9e8-11e9-b44d-005056affe3f_0
bfd4c07afe64        3ce97e5c7e3a                                              "kubectl apply -f /e…"   10 minutes ago      Exited (1) 10 minutes ago                       k8s_rke-network-plugin-pod_rke-network-plugin-deploy-job-x8wjr_kube-system_db99b88b-b9e8-11e9-b44d-005056affe3f_0
555cfb2aa903        192.168.10.85:5000/rancher/pause:3.1                      "/pause"                 10 minutes ago      Exited (0) 10 minutes ago                       k8s_POD_rke-network-plugin-deploy-job-x8wjr_kube-system_db99b88b-b9e8-11e9-b44d-005056affe3f_0
217d27e3beaa        192.168.10.85:5000/rancher/hyperkube:v1.14.5-rancher1     "/opt/rke-tools/entr…"   42 minutes ago      Up 42 minutes                                   kube-proxy
ff1cb63357e1        192.168.10.85:5000/rancher/hyperkube:v1.14.5-rancher1     "/opt/rke-tools/entr…"   42 minutes ago      Up 42 minutes                                   kubelet
3f632eb89e53        192.168.10.85:5000/rancher/hyperkube:v1.14.5-rancher1     "/opt/rke-tools/entr…"   42 minutes ago      Up 42 minutes                                   kube-scheduler
e593aef9d86a        192.168.10.85:5000/rancher/hyperkube:v1.14.5-rancher1     "/opt/rke-tools/entr…"   42 minutes ago      Up 42 minutes                                   kube-controller-manager
71a5be14c5c1        192.168.10.85:5000/rancher/rancher-agent:v2.2.7           "run.sh --server htt…"   42 minutes ago      Up 42 minutes                                   confident_mahavira
6ce5fa9ac584        192.168.10.85:5000/rancher/hyperkube:v1.14.5-rancher1     "/opt/rke-tools/entr…"   42 minutes ago      Up 42 minutes                                   kube-apiserver
64a5691eddfb        192.168.10.85:5000/rancher/rke-tools:v0.1.40              "/bin/bash"              42 minutes ago      Created                                         service-sidekick
b43a2d7e4120        192.168.10.85:5000/rancher/coreos-etcd:v3.3.10-rancher1   "/usr/local/bin/etcd…"   42 minutes ago      Up 42 minutes                                   etcd
75d58e34d888        192.168.10.85:5000/rancher/rke-tools:v0.1.40              "/bin/bash"              43 minutes ago      Exited (0) 43 minutes ago                       cluster-state-deployer
4cfff16224e0        192.168.10.85:5000/rancher/rancher-agent:v2.2.7           "run.sh -- share-roo…"   43 minutes ago      Exited (0) 42 minutes ago                       share-mnt
0a4263574a12        192.168.10.85:5000/rancher/rancher-agent:v2.2.7           "run.sh --server htt…"   43 minutes ago      Up 43 minutes                                   optimistic_fermat
```

Please let me know if you need inspect or logs for some of them.

**Comment by superseb** [comment]
Regarding this specific error message, please share the output of:

```docker logs $(docker ps -qa --filter=label=io.kubernetes.container.name=rke-network-plugin-pod)```

In this case, `docker logs kubelet` will also be helpful.

**Comment by kenny404** [comment]
mmm... a temporary solution, launch this script : docker_run_rancher.sh with
**while [ 1 ];do sh docker_run_rancher.sh;docker ps;sleep 20;done**

it works ?


[docker_run_rancher.zip](https://github.com/rancher/rancher/files/3482695/docker_run_rancher.zip)

**Comment by turicumoperarius** [comment]
Thank for the additional hints. In the kublet container entries with the following statements were present.

`W0809 12:49:09.825091    6833 kubelet_node_status.go:502] Failed to set some node status fields: can't get ip address of node ubuntrancher. error: no default routes found in "/proc/net/route" or "/proc/net/ipv6_ro`

I added a default route by using `sudo route add default gw 192.168.10.238` and now also kublet provisioned after some waiting time.

I also solved the problem with the agent container, which missed the "--address" argument and resatarted all the time, by gettings its "token" and the other starting arguments ("--no-register --only-write-certs") and started a new container by using the main node container start statement with replaced arguments and "token".

After fixing the routing and the start of the proper container, the air gapped single node installation is now working as expected.

Thanks again for the help.
